src/cython_cmake/cmake/UseCython.cmake

```diff
@@ -140,190 +72,120 @@
-  if(NOT _target_language)
-    if("C" IN_LIST _languages)
-      set(_target_language "C")
-    elseif("CXX" IN_LIST _languages)
-      set(_target_language "CXX")
-    endif()
-  else()
-    if(NOT _target_language MATCHES "^(C|CXX)$")
-      message(FATAL_ERROR "TARGET_LANGUAGE must be one of C or CXX")
-    endif()
+  # Support
+  if(NOT CYTHON_CYTHON_ARGS AND DEFINED CYTHON_ARGS)
+    set(CYTHON_CYTHON_ARGS "${CYTHON_ARGS}")
   endif()
 
-  set(_target_language_C_arg "")
-  set(_target_language_C_extension "c")
-  set(_target_language_CXX_arg "--cplus")
-  set(_target_language_CXX_extension "cxx")
-
-  set(_target_language_arg ${_target_language_${_target_language}_arg})
-  set(_target_language_extension ${_target_language_${_target_language}_extension})
-
-  # Set language level
-  set(_language_level ${_args_LANGUAGE_LEVEL})
-  if(NOT _args_LANGUAGE_LEVEL)
-    set(_language_level "3str")
-  else()
-    if(NOT _language_level MATCHES "^(2|3|3str)$")
-      message(FATAL_ERROR "LANGUAGE_LEVEL must be one of 2, 3, or 3str")
-    endif()
+  # Output variable only if set
+  if(CYTHON_OUTPUT_VARIABLE)
+    set(${CYTHON_OUTPUT_VARIABLE} "${CYTHON_OUTPUT}" PARENT_SCOPE)
   endif()
 
-  set(_language_level_2_arg "-2")
-  set(_language_level_3_arg "-3")
-  set(_language_level_3str_arg "--3str")
-
-  set(_language_level_arg "${_language_level_${_language_level}_arg}")
-
-  # Generated depfile is expected to have the ".dep" extension and be located along
-  # side the generated source file.
-  set(_depfile ${generated_file}.dep)
-  set(_depfile_arg "-M")
-
-  set(generated_files)
-
-  foreach(_source_file IN LISTS _source_files)
-    cmake_path(GET _source_file STEM _name)
-    set(generated_file "${CMAKE_CURRENT_BINARY_DIR}/${_name}.${_target_language_extension}")
-    set_source_files_properties(${generated_file} PROPERTIES GENERATED TRUE)
-
-    file(RELATIVE_PATH generated_file_relative
-        ${CMAKE_BINARY_DIR} ${generated_file})
-
-    set(comment "Generating ${_target_language} source ${generated_file_relative}")
-
-    get_source_file_property(pyx_location ${_source_file} LOCATION)
-
-    # Add the command to run the compiler.
-    add_custom_command(
-      OUTPUT ${generated_file}
-      COMMAND ${CYTHON_EXECUTABLE}
-      ARGS
-        ${_target_language_arg}
-        ${_language_level_arg}
-        ${_args_CYTHON_ARGS}
-        ${_depfile_arg}
-        ${pyx_location}
-        --output-file ${generated_file}
-      DEPENDS
-        ${_source_file}
-      DEPFILE
-        ${_cython_depfile}
-      VERBATIM
-      COMMENT ${comment}
-    )
-    list(APPEND generated_files ${generated_file})
-  endforeach()
-
-  if(_args_OUTPUT_VAR)
-    set(_output_var ${_args_OUTPUT_VAR})
-    set(${_output_var} ${generated_files} PARENT_SCOPE)
-  endif()
+  # Generated depfile is expected to have the ".dep" extension and be located
+  # along side the generated source file.
+  set(depfile_path "${CYTHON_OUTPUT}.dep")
```

Comment:
We added depfile support in https://github.com/rapidsai/rapids-cmake/pull/579/files, so you may want to double-check there to see if we did anything differently from what you did here (good or bad) since both our code and the new code in this repo are very stripped down relative to what UseCython.cmake looked like before this PR.

Reply:
All features have been added while maintaining the history.

For reference, depfile support has indeed been added through https://github.com/scikit-build/cython-cmake/commit/490f1998147af498793f88365b2a9293311a74ee

Reply:
The only thing I saw I commented on that PR; that one is missing the depfile as output, while I think it's correct to list it there (since it's not always output, just usually, so CMake likely can't assume it's output).